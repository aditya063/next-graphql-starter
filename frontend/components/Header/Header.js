import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledLogo = styled.h1`
  color: ${props => props.theme.red};
`;

class Header extends React.PureComponent {
  render() {
    return (
      <header>
        <StyledLogo>next-graphql-starter</StyledLogo>
        {this.props.children}
      </header>
    );
  }
}

Header.propTypes = {
  children: PropTypes.node.isRequired
};

export default Header;
