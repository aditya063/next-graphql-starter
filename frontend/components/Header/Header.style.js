import styled from 'styled-components';

import { signOutBtn } from '../SignOut/SignOut.style';

export const header = styled.header`
  background-color: ${props => props.theme.color.blue};
  border-bottom: 1px solid ${props => props.theme.color.black};
  display: grid;
  grid-area: header;
  grid-gap: 10px;
  grid-template-areas: 'header-left header-right';
  padding: 10px 0 0 0;
  place-items: center center;

  > h1 {
    color: ${props => props.theme.color.yellow_bright};
    font-family: 'Play', sans-serif;
    font-size: 1.5rem;
    justify-self: start;
    letter-spacing: 0.05rem;
    margin: 0 0 0 65px;
    text-shadow: 1px 1px ${props => props.theme.color.black};
  }

  > ${signOutBtn} {
    justify-self: end;
    margin-right: 10px;
  }

  @media screen and (min-width: 600px) {
    padding: 10px 20px 0 20px;

    > h1 {
      margin-left: 70px;
    }

    > ${signOutBtn} {
      margin-right: 0;
    }
  }
`;
