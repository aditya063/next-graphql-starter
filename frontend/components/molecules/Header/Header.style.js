import styled from 'styled-components';
import { title } from '../../../constants';
import { Title } from '../../atoms/HeaderTitle/HeaderTitle.style';

export const Header = styled.div`
  display: grid;
  grid-template-areas:
    'top-left top-right'
    'bottom bottom';
  grid-template-columns: 1fr auto;
  align-items: center;
  flex-grow: 1;

  background-color: ${props => props.theme.color.lightBlue};
  border-bottom: 1px solid ${props => props.theme.color.black};
  padding: 10px 10px 0 10px;

  /* display: flex; */
  /* flex-direction: column; */

  /* flex-grow: 1; */
  /* padding: 10px; */
  /* justify-content: space-between; */
  /* align-items: center; */

  /* @media screen and (min-width: 920px) {
    padding: 10px 0;
  } */
`;

export const HeaderTitle = styled(Title)`
  padding-left: ${props => props.children === title && '70px'};
`;

export const Bottom = styled.div`
  grid-area: bottom;

  padding-bottom: ${props => !props.children && '10px'};
`;

// export const HeaderTitle = styled(Title)`
//   /* @media screen and (min-width: 420px) {
//     margin-left: 70px;
//   } */
// `;

//topmain

// h1 {
//   margin: 5px 0 5px 80px;
// }

// > ${SignOutBtn} {
//   grid-area: top-right;
//   /* margin-right: 10px; */
// }

// > ${IdeaCardForm} {
//   grid-area: bottom;
// }

// @media screen and (min-width: 920px) {
//   > ${SignOutBtn} {
//     /* margin-right: 0; */
//   }
// }
// `;

// export const TopDetail = styled.div`
//   /* display: flex;
//   flex-grow: 1;
//   padding: 10px;
//   justify-content: space-between;
//   align-items: center; */

//   background-color: ${props => props.theme.color.lightBlue};
//   border-bottom: 1px solid ${props => props.theme.color.black};
//   padding: 10px;

//   display: grid;
//   grid-template-areas: 'top-left top-right';
//   grid-template-columns: 1fr auto;
//   align-items: center;
//   flex-grow: 1;

//   @media screen and (min-width: 920px) {
//     padding: 10px 0;
//   }
// `;

// export const HeaderTitle = styled(Title)`
//   /* @media screen and (min-width: 420px) {
//     margin-left: 70px;
//   } */
// `;

/* > h1 {
    margin: 5px 0 5px 80px;
  } */

/* > ${SignOutBtn} {
    grid-area: top-right;
    margin-right: 10px;
  } */

/* > ${IdeaCardForm} {
    grid-area: bottom;
  } */

/* @media screen and (min-width: 920px) {
    > ${SignOutBtn} {
      margin-right: 0;
    }
  } */

//   h1 {
//     font-family: 'Play', sans-serif;
//     letter-spacing: 0.05rem;
//   }
// `;
