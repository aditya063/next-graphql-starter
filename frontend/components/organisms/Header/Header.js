import PropTypes from 'prop-types';
import SignOutBtn from '../../SignOutBtn/SignOutBtn';
import { title } from '../../../constants';
import * as sc from './Header.style';

const Header = props => {
  return (
    <sc.Header>
      <sc.HeaderTitle>{props.ideaId ? props.ideaId : title}</sc.HeaderTitle>

      <SignOutBtn />

      <sc.Bottom>{props.children}</sc.Bottom>
    </sc.Header>
  );
};

Header.propTypes = {
  children: PropTypes.element,
  ideaId: PropTypes.string
};

export default React.memo(Header);
