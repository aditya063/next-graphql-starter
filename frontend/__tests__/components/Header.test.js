import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { ThemeProvider } from 'styled-components';
import Header from '../../components/Header/Header';
import theme from '../../styles/theme.style';

describe('Header', () => {
  let mockProps;
  let mockQueries;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    mockQueries = [];
    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <ThemeProvider theme={theme}>
          <Header {...mockProps} />
        </ThemeProvider>
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('matches snapshot', () => {
    const wrapSnap = wrapper.find({ snapshot: 'Header' });

    expect(wrapSnap).toMatchSnapshot();
  });
});
