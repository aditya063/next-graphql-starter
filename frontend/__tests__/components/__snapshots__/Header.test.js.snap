// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
Array [
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  grid-column: span 2;
  position: relative;
}

.c2 > img {
  width: 50px;
  position: absolute;
  top: -30px;
  left: 10px;
}

.c2 > textarea {
  padding: 10px;
  outline: none;
  min-height: 56px;
  width: 100%;
  resize: vertical;
  border: none;
  margin-right: 1px;
}

.c2 > button {
  border: none;
  outline: none;
  cursor: pointer;
  background-color: #fffacd;
  padding: 0;
}

.c2 > button:active {
  background-color: #FFFF99;
}

.c0 {
  grid-area: header;
  display: grid;
  grid-gap: 10px;
  background-color: #393939;
  grid-template-columns: 1fr 1fr;
  grid-template-areas: 'header-left header-right';
  place-items: center center;
  padding: 10px 0 1px 0;
}

.c1 {
  font-size: 1.5rem;
  font-family: 'Play',sans-serif;
  color: #FFFF99;
  margin: 0 0 0 65px;
  justify-self: start;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

@media screen and (min-width:600px) {
  .c2 > img {
    left: 0;
  }
}

@media screen and (min-width:600px) {
  .c0 {
    padding: 10px 20px 1px 20px;
  }
}

@media screen and (min-width:600px) {
  .c1 {
    margin-left: 70px;
  }
}

<Headerstyle__header
    snapshot="Header"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Headerstyle__header-sc-6zuvh-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "grid-area:header;display:grid;grid-gap:10px;background-color:",
              [Function],
              ";grid-template-columns:1fr 1fr;grid-template-areas:'header-left header-right';place-items:center center;padding:10px 0 1px 0;@media screen and (min-width:600px){padding:10px 20px 1px 20px;}",
            ],
          },
          "displayName": "Headerstyle__header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Headerstyle__header-sc-6zuvh-0",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      snapshot="Header"
    >
      <header
        className="c0"
      >
        <Headerstyle__h1>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Headerstyle__h1-sc-6zuvh-1",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "font-size:1.5rem;font-family:'Play',sans-serif;color:",
                    [Function],
                    ";margin:0 0 0 65px;justify-self:start;letter-spacing:0.05rem;@media screen and (min-width:600px){margin-left:70px;}",
                  ],
                },
                "displayName": "Headerstyle__h1",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Headerstyle__h1-sc-6zuvh-1",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h1
              className="c1"
            >
              Starter
            </h1>
          </StyledComponent>
        </Headerstyle__h1>
        <SignOn>
          <Component>
            <Query
              errorPolicy="all"
              query={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "ME_QUERY",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "me",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "email",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 60,
                    "start": 0,
                  },
                }
              }
            >
              <Component>
                <p
                  snapshot="DisplayLoading"
                >
                  Loading ...
                </p>
              </Component>
            </Query>
          </Component>
        </SignOn>
        <IdeaCardForm>
          <Mutation
            errorPolicy="all"
            mutation={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "CREATE_IDEA_MUTATION",
                    },
                    "operation": "mutation",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "content",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "content",
                                },
                              },
                            },
                          ],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "createIdea",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [
                      Object {
                        "defaultValue": undefined,
                        "directives": Array [],
                        "kind": "VariableDefinition",
                        "type": Object {
                          "kind": "NonNullType",
                          "type": Object {
                            "kind": "NamedType",
                            "name": Object {
                              "kind": "Name",
                              "value": "String",
                            },
                          },
                        },
                        "variable": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "content",
                          },
                        },
                      },
                    ],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 109,
                  "start": 0,
                },
              }
            }
            onError={[Function]}
            refetchQueries={
              Array [
                Object {
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": Object {
                          "kind": "Name",
                          "value": "ME_IDEAS_QUERY",
                        },
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "getUserIdeas",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "content",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 78,
                      "start": 0,
                    },
                  },
                },
              ]
            }
            variables={
              Object {
                "content": "",
              }
            }
          >
            <IdeaCardFormstyle__form
              onSubmit={[Function]}
              snapshot="IdeaCardForm"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "IdeaCardFormstyle__form-sc-4er11-0",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "display:flex;width:100%;grid-column:span 2;position:relative;> img{width:50px;position:absolute;top:-30px;left:10px;}> textarea{padding:10px;outline:none;min-height:56px;width:100%;resize:vertical;border:none;margin-right:1px;}> button{border:none;outline:none;cursor:pointer;background-color:",
                        [Function],
                        ";padding:0;&:active{background-color:",
                        [Function],
                        ";}}@media screen and (min-width:600px){> img{left:0;}}",
                      ],
                    },
                    "displayName": "IdeaCardFormstyle__form",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "IdeaCardFormstyle__form-sc-4er11-0",
                    "target": "form",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                onSubmit={[Function]}
                snapshot="IdeaCardForm"
              >
                <form
                  className="c2"
                  onSubmit={[Function]}
                >
                  <img
                    alt="ideabox"
                    src="static/ideabox.png"
                  />
                  <textarea
                    name="idea"
                    onChange={[Function]}
                    placeholder="What's on your mind?"
                    type="text"
                    value=""
                  />
                  <button
                    disabled={true}
                    type="submit"
                  >
                    Add Idea
                  </button>
                </form>
              </StyledComponent>
            </IdeaCardFormstyle__form>
          </Mutation>
        </IdeaCardForm>
      </header>
    </StyledComponent>
  </Headerstyle__header>,
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  grid-column: span 2;
  position: relative;
}

.c2 > img {
  width: 50px;
  position: absolute;
  top: -30px;
  left: 10px;
}

.c2 > textarea {
  padding: 10px;
  outline: none;
  min-height: 56px;
  width: 100%;
  resize: vertical;
  border: none;
  margin-right: 1px;
}

.c2 > button {
  border: none;
  outline: none;
  cursor: pointer;
  background-color: #fffacd;
  padding: 0;
}

.c2 > button:active {
  background-color: #FFFF99;
}

.c0 {
  grid-area: header;
  display: grid;
  grid-gap: 10px;
  background-color: #393939;
  grid-template-columns: 1fr 1fr;
  grid-template-areas: 'header-left header-right';
  place-items: center center;
  padding: 10px 0 1px 0;
}

.c1 {
  font-size: 1.5rem;
  font-family: 'Play',sans-serif;
  color: #FFFF99;
  margin: 0 0 0 65px;
  justify-self: start;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

@media screen and (min-width:600px) {
  .c2 > img {
    left: 0;
  }
}

@media screen and (min-width:600px) {
  .c0 {
    padding: 10px 20px 1px 20px;
  }
}

@media screen and (min-width:600px) {
  .c1 {
    margin-left: 70px;
  }
}

<StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Headerstyle__header-sc-6zuvh-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "grid-area:header;display:grid;grid-gap:10px;background-color:",
            [Function],
            ";grid-template-columns:1fr 1fr;grid-template-areas:'header-left header-right';place-items:center center;padding:10px 0 1px 0;@media screen and (min-width:600px){padding:10px 20px 1px 20px;}",
          ],
        },
        "displayName": "Headerstyle__header",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Headerstyle__header-sc-6zuvh-0",
        "target": "header",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    snapshot="Header"
  >
    <header
      className="c0"
    >
      <Headerstyle__h1>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Headerstyle__h1-sc-6zuvh-1",
                "isStatic": false,
                "lastClassName": "c1",
                "rules": Array [
                  "font-size:1.5rem;font-family:'Play',sans-serif;color:",
                  [Function],
                  ";margin:0 0 0 65px;justify-self:start;letter-spacing:0.05rem;@media screen and (min-width:600px){margin-left:70px;}",
                ],
              },
              "displayName": "Headerstyle__h1",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Headerstyle__h1-sc-6zuvh-1",
              "target": "h1",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <h1
            className="c1"
          >
            Starter
          </h1>
        </StyledComponent>
      </Headerstyle__h1>
      <SignOn>
        <Component>
          <Query
            errorPolicy="all"
            query={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "ME_QUERY",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 60,
                  "start": 0,
                },
              }
            }
          >
            <Component>
              <p
                snapshot="DisplayLoading"
              >
                Loading ...
              </p>
            </Component>
          </Query>
        </Component>
      </SignOn>
      <IdeaCardForm>
        <Mutation
          errorPolicy="all"
          mutation={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "CREATE_IDEA_MUTATION",
                  },
                  "operation": "mutation",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "content",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "content",
                              },
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createIdea",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [
                    Object {
                      "defaultValue": undefined,
                      "directives": Array [],
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "String",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                      },
                    },
                  ],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 109,
                "start": 0,
              },
            }
          }
          onError={[Function]}
          refetchQueries={
            Array [
              Object {
                "query": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "ME_IDEAS_QUERY",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "getUserIdeas",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "content",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 78,
                    "start": 0,
                  },
                },
              },
            ]
          }
          variables={
            Object {
              "content": "",
            }
          }
        >
          <IdeaCardFormstyle__form
            onSubmit={[Function]}
            snapshot="IdeaCardForm"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "IdeaCardFormstyle__form-sc-4er11-0",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "display:flex;width:100%;grid-column:span 2;position:relative;> img{width:50px;position:absolute;top:-30px;left:10px;}> textarea{padding:10px;outline:none;min-height:56px;width:100%;resize:vertical;border:none;margin-right:1px;}> button{border:none;outline:none;cursor:pointer;background-color:",
                      [Function],
                      ";padding:0;&:active{background-color:",
                      [Function],
                      ";}}@media screen and (min-width:600px){> img{left:0;}}",
                    ],
                  },
                  "displayName": "IdeaCardFormstyle__form",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "IdeaCardFormstyle__form-sc-4er11-0",
                  "target": "form",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onSubmit={[Function]}
              snapshot="IdeaCardForm"
            >
              <form
                className="c2"
                onSubmit={[Function]}
              >
                <img
                  alt="ideabox"
                  src="static/ideabox.png"
                />
                <textarea
                  name="idea"
                  onChange={[Function]}
                  placeholder="What's on your mind?"
                  type="text"
                  value=""
                />
                <button
                  disabled={true}
                  type="submit"
                >
                  Add Idea
                </button>
              </form>
            </StyledComponent>
          </IdeaCardFormstyle__form>
        </Mutation>
      </IdeaCardForm>
    </header>
  </StyledComponent>,
]
`;
